pkg/proc/target_exec.go

```diff
@@ -797,14 +782,47 @@ func setStepIntoBreakpoints(dbp *Target, curfn *Function, text []AsmInstruction,
 			}
 		} else {
 			// Non-absolute call instruction, set a StepBreakpoint here
-			if _, err := allowDuplicateBreakpoint(dbp.SetBreakpoint(0, instr.Loc.PC, StepBreakpoint, sameGCond)); err != nil {
+			bp, err := allowDuplicateBreakpoint(dbp.SetBreakpoint(0, instr.Loc.PC, StepBreakpoint, sameGCond))
+			if err != nil {
 				return err
 			}
+			breaklet := bp.Breaklets[len(bp.Breaklets)-1]
+			breaklet.callback = stepIntoCallback
 		}
 	}
 	return nil
 }
 
+// stepIntoCallback is a callback called when a StepBreakpoint is hit, it
+// disassembles the current instruction to figure out its destination and
+// sets a breakpoint on it.
+func stepIntoCallback(curthread Thread, grp *TargetGroup, p *Target) (bool, error) {
+	if grp.GetDirection() != Forward {
```

Comment:
Couldn't we use `p.recman.GetDirection()` here instead of passing the group in?

Reply:
Done.